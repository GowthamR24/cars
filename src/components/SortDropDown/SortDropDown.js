import React from 'react';
import { Dropdown, DropdownButton } from 'react-bootstrap';

const dropdownoptions = [
  {
    value: 'make',
    type: 'asc',
    label: 'Make A-Z',
  },
  {
    value: 'make',
    type: 'desc',
    label: 'Make Z-A',
  },
  {
    value: 'model',
    type: 'asc',
    label: 'Model A-Z',
  },
  {
    value: 'model',
    type: 'desc',
    label: 'Model Z-A',
  },
  {
    value: 'color',
    type: 'asc',
    label: 'Color A-Z',
  },
  {
    value: 'color',
    type: 'desc',
    label: 'Color Z-A',
  },
  {
    value: 'year',
    type: 'asc',
    label: 'Year Low to High',
  },
  {
    value: 'year',
    type: 'desc',
    label: 'Year High to Low',
  },
];

const SortDropDown = ({ handleSort, disabled }) => {
  return (
    <DropdownButton
      variant='outline-secondary'
      title='Sort By'
      disabled={disabled}
    >
      {dropdownoptions.map((option) => (
        <Dropdown.Item
          key={option.label}
          value={option.value}
          onClick={() => handleSort(option.value, option.type)}
        >
          {option.label}
        </Dropdown.Item>
      ))}
    </DropdownButton>
  );
};

export default SortDropDown;
