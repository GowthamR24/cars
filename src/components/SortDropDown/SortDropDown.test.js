import React from 'react';
import { mount } from 'enzyme';
import SortDropDown from './SortDropDown';

describe('SortDropDown', () => {
  let wrapper;
  const props = {
    handleSort: jest.fn(),
    disabled: false,
  };

  beforeEach(() => {
    wrapper = mount(<SortDropDown {...props} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when disabled', () => {
    wrapper.setProps({ disabled: true });
    expect(wrapper).toMatchSnapshot();
  });
});
