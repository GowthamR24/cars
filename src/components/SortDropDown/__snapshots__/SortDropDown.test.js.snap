// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortDropDown should render correctly 1`] = `ReactWrapper {}`;

exports[`SortDropDown should render correctly when disabled 1`] = `ReactWrapper {}`;
